api/infrastructure/parallelcluster-api.yaml

```diff
@@ -0,0 +1,246 @@
+  UsesOnlyCustomDomain: !And
+    - !Condition UsesCustomDomain
+    - !Not [!Condition UsesRoute53Configuration]
+  UsesCustomDomainAndRoute53Configuration: !And
+    - !Condition UsesCustomDomain
+    - !Condition UsesRoute53Configuration
+  DoesNotUseCustomDomain: !Not [!Condition UsesCustomDomain]
+
+
+Resources:
+
+  # We need to define three AWS::Serverless::Api due to an issue with the handling of AWS::NoValue
+  # See related GitHub issue: https://github.com/aws/serverless-application-model/issues/1435
+
+  ApiGatewayApiWithOnlyCustomDomain:
+    Condition: UsesOnlyCustomDomain
+    Type: AWS::Serverless::Api
+    Properties:
+      Tags:
+        parallelcluster: !Ref ParallelClusterVersion
+      StageName: prod
+      DefinitionBody:
+        Fn::Transform:
+          Name: AWS::Include
+          Parameters:
+            Location: !Ref ApiDefinitionS3Uri
+      Domain:
+        DomainName: !Ref CustomDomainName
+        CertificateArn: !Ref CustomDomainCertificate
+
+  ApiGatewayApiWithCustomDomainAndRoute53Configuration:
+    Condition: UsesCustomDomainAndRoute53Configuration
+    Type: AWS::Serverless::Api
+    Properties:
+      Tags:
+        parallelcluster: !Ref ParallelClusterVersion
+      StageName: prod
+      DefinitionBody:
+        Fn::Transform:
+          Name: AWS::Include
+          Parameters:
+            Location: !Ref ApiDefinitionS3Uri
+      Domain:
+        DomainName: !Ref CustomDomainName
+        CertificateArn: !Ref CustomDomainCertificate
+        Route53:
+          HostedZoneId: !Ref CustomDomainHostedZoneId
+
+  ApiGatewayApiWithoutCustomDomain:
+    Condition: DoesNotUseCustomDomain
+    Type: AWS::Serverless::Api
+    Properties:
+      Tags:
+        parallelcluster: !Ref ParallelClusterVersion
+      StageName: prod
+      DefinitionBody:
+        Fn::Transform:
+          Name: AWS::Include
+          Parameters:
+            Location: !Ref ApiDefinitionS3Uri
+
+  APIGatewayExecutionRole:
+   Type: AWS::IAM::Role
+   Properties:
+     AssumeRolePolicyDocument:
+       Version: 2012-10-17
+       Statement:
+         - Effect: Allow
+           Principal:
+             Service:
+               - apigateway.amazonaws.com
+           Action:
+             - 'sts:AssumeRole'
+     Policies:
+       - PolicyName: lambda-invoke
+         PolicyDocument:
+           Version: '2012-10-17'
+           Statement:
+             - Effect: Allow
+               Action: lambda:InvokeFunction
+               Resource: !GetAtt ParallelClusterFunction.Arn
+
+  ParallelClusterFunction:
+    Type: AWS::Serverless::Function
+    DependsOn: LambdaLogGroup
+    Properties:
+      PackageType: Image
+      Tracing: Active
+      Tags:
+        parallelcluster: !Ref ParallelClusterVersion
+      Policies:
+        - AWSLambdaBasicExecutionRole
+        - AWSXRayDaemonWriteAccess
+        - AdministratorAccess
+      ImageUri: !If [UsesCustomEcrImageUri, !Ref EcrImageUri, !Ref PublicEcrImageUri]
```

Comment:
In the else clause we probably want to use the imported image not the public one